Resending pull request

wzod opened the pull request:
Resending as previous pull request was closed (should not have been).

==========
Updated the script with the following changes:

-    bring yara up to version 3.2.0
-    add install of unzip in the 'extract' function since default Ubuntu Server does not include the package (ensure distorm.zip gets extracted)
-    subsitute downloading ipython from project site to using apt repo for installation (there were some dependency issues/errors when initiating ipython on my default ubuntu server install)
-    add ipython-notebook (in case folks want to interact with ipython via a browser)

4n6k commented on 4n6k_volatility_installer.sh:
```diff
@@ -115,6 +114,7 @@ verify() {
 
 # Extract the downloaded archives
 extract() {
+  sudo apt-get install unzip 
```
"-y --force-yes" on this

wzod replied:
Nice catch!

wzod commented:
Reverted the changes re: ipython/ipython notebook; tested and looks good on my end.

Also included apt-get update prior to installing unzip to ensure no issues when installing unzip (had a case during testing where I couldn't connect to the apt repo until after completing an apt-get update.

4n6k commented:
Wound up making some additional changes, as well, so just implemented them all at once without the merge. Gonna close this; thanks again! (p.s. you're in the commit log and on Twitter; thank you!)

4n6k commented:
Hey Zod, went ahead and fixed up a bunch of stuff including your changes.
v1.1.2 is live! YARA was actually updated in the last few days to 3.3.0 (so
I added THAT one instead of 3.2.0. Wound up not merging, as I was changing
some other stuff as well. But you are thanked in the commit logs and on
Twitter. Let me know if you are fine with this -- I'd like to give credit
where it is due, so let me know if this is sufficient for you.

Keep me updated on the docker stuff!

Thanks again

-Dan



On Sun, Feb 8, 2015 at 4:21 PM, wzod <notifications@github.com> wrote:

> Reverted the changes re: ipython/ipython notebook; tested and looks good
> on my end.
>
> Also included apt-get update prior to installing unzip to ensure no issues
> when installing unzip (had a case during testing where I couldn't connect
> to the apt repo until after completing an apt-get update.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/4n6k/4n6k_volatility_installer/pull/2#issuecomment-73435777>
> .
>



-- 
Dan Pullega (@4n6k <https://twitter.com/4n6k>)
4n6k.com

wzod commented:
Nice work, Dan!  Glad you were able to include the updated version of YARA
as well as the other changes; totally understand the need to not merge
since the pull request was deprecated by events.

Regarding the changes, I tested on my base Ubuntu Server 14.04 install and
noticed the following error from the install.log:

4n6k_volatility_installer/4n6k_volatility_installer.sh: line 139: cd:
distorm3: No such file or directory

I think this might be due to a timing issue as the base Ubuntu Server
install does not include unzip.  The workaround was including the install
of unzip in the extract function so that distorm unpacks and loads
successfully.  Would it makes sense to include the install of unzip in an
earlier step in the script?

Working on completing the docker image today; I'll definitely keep you
posted as that gets close to completion.

Thanks!


On Sat, Feb 14, 2015 at 11:01 PM, 4n6k <notifications@github.com> wrote:

> Hey Zod, went ahead and fixed up a bunch of stuff including your changes.
> v1.1.2 is live! YARA was actually updated in the last few days to 3.3.0 (so
> I added THAT one instead of 3.2.0. Wound up not merging, as I was changing
> some other stuff as well. But you are thanked in the commit logs and on
> Twitter. Let me know if you are fine with this -- I'd like to give credit
> where it is due, so let me know if this is sufficient for you.
>
> Keep me updated on the docker stuff!
>
> Thanks again
>
> -Dan
>
>
>
> On Sun, Feb 8, 2015 at 4:21 PM, wzod <notifications@github.com> wrote:
>
> > Reverted the changes re: ipython/ipython notebook; tested and looks good
> > on my end.
> >
> > Also included apt-get update prior to installing unzip to ensure no
> issues
> > when installing unzip (had a case during testing where I couldn't connect
> > to the apt repo until after completing an apt-get update.
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
> https://github.com/4n6k/4n6k_volatility_installer/pull/2#issuecomment-73435777
> >
> > .
> >
>
>
>
> --
> Dan Pullega (@4n6k <https://twitter.com/4n6k>)
> 4n6k.com
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/4n6k/4n6k_volatility_installer/pull/2#issuecomment-74403124>
> .
>



-- 
-Zod
@wzod

4n6k commented:
Hmmm. I include the install of unzip when the program flow hits the
apt_get_install() function (which should be one if the first things the
script executes). But let me test and see what the deal is. Thanks for the
heads up. I thought the unzip inclusion fixed that but I'll have to check
again

~Sent succinctly via mobile
On Feb 15, 2015 8:29 AM, "Zod" <notifications@github.com> wrote:

> Nice work, Dan! Glad you were able to include the updated version of YARA
> as well as the other changes; totally understand the need to not merge
> since the pull request was deprecated by events.
>
> Regarding the changes, I tested on my base Ubuntu Server 14.04 install and
> noticed the following error from the install.log:
>
> 4n6k_volatility_installer/4n6k_volatility_installer.sh: line 139: cd:
> distorm3: No such file or directory
>
> I think this might be due to a timing issue as the base Ubuntu Server
> install does not include unzip. The workaround was including the install
> of unzip in the extract function so that distorm unpacks and loads
> successfully. Would it makes sense to include the install of unzip in an
> earlier step in the script?
>
> Working on completing the docker image today; I'll definitely keep you
> posted as that gets close to completion.
>
> Thanks!
>
>
> On Sat, Feb 14, 2015 at 11:01 PM, 4n6k <notifications@github.com> wrote:
>
> > Hey Zod, went ahead and fixed up a bunch of stuff including your changes.
> > v1.1.2 is live! YARA was actually updated in the last few days to 3.3.0
> (so
> > I added THAT one instead of 3.2.0. Wound up not merging, as I was
> changing
> > some other stuff as well. But you are thanked in the commit logs and on
> > Twitter. Let me know if you are fine with this -- I'd like to give credit
> > where it is due, so let me know if this is sufficient for you.
> >
> > Keep me updated on the docker stuff!
> >
> > Thanks again
> >
> > -Dan
> >
> >
> >
> > On Sun, Feb 8, 2015 at 4:21 PM, wzod <notifications@github.com> wrote:
> >
> > > Reverted the changes re: ipython/ipython notebook; tested and looks
> good
> > > on my end.
> > >
> > > Also included apt-get update prior to installing unzip to ensure no
> > issues
> > > when installing unzip (had a case during testing where I couldn't
> connect
> > > to the apt repo until after completing an apt-get update.
> > >
> > > —
> > > Reply to this email directly or view it on GitHub
> > > <
> >
> https://github.com/4n6k/4n6k_volatility_installer/pull/2#issuecomment-73435777
> > >
> > > .
> > >
> >
> >
> >
> > --
> > Dan Pullega (@4n6k <https://twitter.com/4n6k>)
> > 4n6k.com
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
> https://github.com/4n6k/4n6k_volatility_installer/pull/2#issuecomment-74403124
> >
> > .
> >
>
>
>
> --
> -Zod
> @wzod
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/4n6k/4n6k_volatility_installer/pull/2#issuecomment-74419485>
> .
>

4n6k commented:
Nope, you're right. That function doesn't hit until things are extracted. Fixing now!

4n6k commented:
Should be all good now. Tested on Ubuntu Server 14.01. Let me know if there are any kinks

wzod commented:
Just tested on a fresh install; looks great!

Thank you!!

On Sun, Feb 15, 2015 at 1:28 PM, 4n6k <notifications@github.com> wrote:

> Should be all good now. Tested on Ubuntu Server 14.01. Let me know if
> there are any kinks
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/4n6k/4n6k_volatility_installer/pull/2#issuecomment-74429663>
> .
>



-- 
-Zod
@wzod
